PYME/recipes/kinetics_recipes.py

```diff
@@ -0,0 +1,428 @@
+        from PYME.Analysis.BleachProfile.kinModels import eimod
+
+        #todo: add higher order fits
+
+        oncol = self.onTimesColName
+        offcol = self.offTimesColName
+
+        inp = namespace[self.inputName]
+
+        if 'mdh' in dir(inp):
+            mdh = inp.mdh
+            cyctime = mdh.getEntry('Camera.CycleTime')
+        else:
+            cyctime = self.exposure_time
+
+        ds = tabular.mappingFilter(inp)
+
+        off_fit_limit_index = np.round(self.maxOffTimeSecondsToFit/cyctime).astype('int')
+
+        offtimeslimit = np.round(self.minOffTimeSecondsToFit/cyctime).astype('int')
+
+        onbins = np.arange(1, max(ds[oncol])+1, 1)
+        offbins = np.arange(offtimeslimit, off_fit_limit_index, np.round(self.offTimesBinSizeInSeconds/cyctime).astype('int'))
+
+
+        n_on, bins_on = np.histogram(ds[oncol], bins=onbins)
+        n_off, bins_off = np.histogram(ds[offcol], bins=offbins)
+
+        bins_on = bins_on[:-1]
+        bins_off = bins_off[:-1]
+
+        res_off = FitModelWeighted(eimod, [n_off[0]*self.off_init_condition_multiple, 1], n_off, 1. / (np.sqrt(n_off) + 1), bins_off * cyctime)
+
+        res_on = FitModelWeighted(eimod, [n_on[1], 1], n_on[1:], 1. / (np.sqrt(n_on[1:]) + 1), bins_on[1:] * cyctime)
+
+        pylab.figure()
+
+        pylab.bar(bins_on*cyctime, n_on, width=np.diff(bins_on)[0]*cyctime, alpha=0.4, fc=colours[0])
+        pylab.plot(np.linspace(bins_on[0], max(bins_on), max(bins_on) * 10) * cyctime,
+                   eimod(res_on[0], np.linspace(bins_on[0], max(bins_on), max(bins_on) * 10) * cyctime), colours[0],
+                   lw=3)
+
+        pylab.ylabel('Events')
+        pylab.xlabel('Event Duration [s]')
+        pylab.xlim((0, np.round(n_on[-1]*1.2).astype('int')))
+        pylab.title('Blink Fluorescence Durations')
+        pylab.figtext(.6, .8 - .05, 'tau = %3.4f' % (res_on[0][1]), size=18, color=colours[1])
+        pylab.show()
+
+
+        pylab.figure()
+
+        pylab.xlim(right=np.round(self.maxOffTimeSecondsToFit/cyctime).astype('int'))
+        pylab.bar(bins_off * cyctime, n_off, width=np.diff(bins_off)[0] * cyctime, alpha=0.3, fc=colours[2])
+        pylab.plot(np.linspace(bins_off[0], max(bins_off), max(bins_off) * 10) * cyctime,
+                   eimod(res_off[0], np.linspace(1, max(bins_off), max(bins_off) * 10) * cyctime),
+                   colours[3], lw=3, alpha=.8)
+        pylab.ylabel('Events')
+        pylab.xlabel('Event Duration [s]')
+        pylab.ylim((.9, pylab.ylim()[1]))
+        pylab.title('Blink dark times')
+
+        pylab.figtext(.6, .8 - .05, 'tau = %3.4f' % (res_off[0][1]), size=18, color=colours[1])
+
+        pylab.show()
+
+        powers = np.array([res_on[0][1], res_off[0][1]])
+        times = np.array([cyctime, (max(ds[oncol]).astype(float)+1)*cyctime, (max(ds[offcol]).astype(float)+1)*cyctime])
+
+        output_list = [powers, times, n_on, n_off]
+
+        namespace[self.outputEmphistData] = output_list
```

Comment:
putting a list as an output doesn't really fit with the recipes data model. Tabular (w/ metata) or ragged would be preferred.

Reply:
This was basically just for use with the save_emphistjson function, so I'll probably just get rid of the save_emphist in favor of hdfoutput. For the json output (for simulating), I'll add a traits Bool input rather than an entirely different module

Reply:
I think it would be cleanest if you always output something that could be saved using an output module if you want to, but could also be loaded by the simulator. In other words, rather than adding a bool flag to allow someone to ~break the data model, would be nicer if they could just load an hdf into the simulator. I haven't looked at that code, but would that be possible?

Reply:
Oh yeah that's definitely possible. Should be easy to enough to have it accept both hdf and json files, I'll include that in my next pr